chain/vm/invoker.go

```diff
@@ -284,17 +284,16 @@ func DecodeParams(b []byte, out interface{}) error {
 }
 
 func DumpActorState(i *ActorRegistry, act *types.Actor, b []byte) (interface{}, error) {
-	// Account & Embryo code special case
-	if builtin.IsAccountActor(act.Code) || builtin.IsEmbryoActor(act.Code) {
-		return nil, nil
-	}
-
 	actInfo, ok := i.actors[act.Code]
 	if !ok {
 		return nil, xerrors.Errorf("state type for actor %s not found", act.Code)
 	}
 
 	um := actInfo.vmActor.State()
+	if um == nil {
+		// TODO: I would like to assert that we have the empty object here
+		return nil, nil
+	}
```

Comment:
Do the embryo actors actually not have state? What does the state cid in actor entry point to?

Reply:
Embryo actors (soon to be renamed to "placeholders"), EthAccounts, and the EAM do not have state. Their `head`s are the [empty object CID](https://github.com/filecoin-project/lotus/blob/0b6d1ad20bd629c9b309262339d400ece81d3840/chain/vm/mkactor.go#L40).

This CID will get put into the blockstore as part of [the migration](https://github.com/filecoin-project/go-state-types/blob/08baf7047877664962ecfefdae09a4f4c7f4741f/builtin/v10/migration/top.go#L354-L361) (if it isn't already).

Reply:
Then I guess, if we’re not using nil state anywhere to mean “actor not found”, this seems fine

Reply:
Good question -- I don't know of any place where we do that, but I'll take a look before assuming.